pkg/ctl/cmdutils/configfile.go

```diff
@@ -56,7 +56,9 @@ func (l *commonClusterConfigLoader) Load() error {
 		return l.validateWithoutConfigFile()
 	}
 
-	if err := eks.LoadConfigFromFile(l.ClusterConfigFile, l.ClusterConfig); err != nil {
+	var err error
+
+	if l.ClusterConfig, err = eks.LoadConfigFromFile(l.ClusterConfigFile); err != nil {
```

Comment:
To be clear, from now on `l.ClusterConfig` will be initialised by explicit assignment, right? In that case, we must make sure that `l.ClusterConfigFile == ""` always returns before we get here, so that `l.ClusterConfig` doesn't get reset. Perhaps it would help to add a comment above to clarify this intention, or restructure the code a little to signify how it's supposed to works, what do you think?

Reply:
Or we can add a test that ensure we don't reset the reference that `l.ClusterConfig` holds when `l.ClusterConfigFile == ""`... perhaps that's the best thing we can do here?

Reply:
>To be clear, from now on l.ClusterConfig will be initialised by explicit assignment, right?

Yes, but only if `ClusterConfigFile` is set.

> In that case, we must make sure that l.ClusterConfigFile == "" always returns before we get here, so that l.ClusterConfig doesn't get reset.

Yes, we do make sure of that.

>Perhaps it would help to add a comment above to clarify this intention, or restructure the code a little to signify how it's supposed to works, what do you think?

Agreed, I'll add a comment.

The change here is that instead of dereferencing the pointer and assigning it the new config in `eks.LoadConfigFromFile`, we have it return the new config and overwrite it at the call site. This would affect code that assigned the ClusterConfig locally instead of accessing it via the `ResourceCmd` reference (`l.ClusterConfig` here) before this call is made, which is only done for setting flags in `createClusterCmd`.


>Or we can add a test that ensures we don't reset the reference that l.ClusterConfig holds when l.ClusterConfigFile == ""... perhaps that's the best thing we can do here?

Should be easy to add a test for this but I believe this isn't something a unit test should test. As long as we test both forms (with and without ClusterConfigFile) and assert the values are set properly, it should cover this change. 

Also, relying on having the contents of the pointer mutated from elsewhere can be error-prone.

Reply:
> Also, relying on having the contents of a pointer mutated from elsewhere can be error-prone.

Yes, but that's currently something we do throughout. Personally, I just think that Go is unhelpful as it doesn't provide ways to prevent mutation and often encourages it. So I just embrace mutation, because it's often the easiest way. Something that would be good to discuss, but I wouldn't make it a priority just yet, it will a very big change if we decide to do something about it.

Reply:
> Agreed, I'll add a comment.

Please do that, and we can merge!

Reply:
We indeed need to do something about this. Mutating parameters is not good practice. Even if the code has other instances of this pattern that doesn't make it a good reason to keep following it. Little by little we have to get rid of that.